CLI/Sources/KnitCodeGen/ConfigurationSet.swift

```diff
@@ -145,3 +145,56 @@ public extension ConfigurationSet {
                 """
 
 }
+
+// MARK: - Validation
+
+/// Validate that there are not duplicate registrations _within_ this ConfigurationSet
+/// which represents a single module.
+/// Note that this will not find duplicate registrations across modules.
+extension ConfigurationSet {
+
+    private struct Key: Hashable {
+        let service: String
```

Comment:
I think both `arguments` and `ifConfigCondition` are needed in this key for uniqueness. ifConfigCondition is a little more complex as per the following code.

```
# if DEBUG
container.register(Service.self, factory: DebugService.init)
#endif

# if RELEASE
container.register(Service.self, factory: ReleaseService.init)
#endif

container.register(Service.self, factory: DuplicateService.init)
```

The DEBUG and RELEASE configurations don't clash (but there's no way to know that from code that those are exclusive) but both clash with the registration that isn't wrapped.

Reply:
Ah thanks yes i'll update

Reply:
Updated to include arguments. Still need to work on the `ifConfigCondition` check as well

Reply:
Also i realized i need to separate the assemblies by `TargetResolver` because we have shadowed registrations for Entities